src/plugins/widgets/topSites/TopSites.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { FC, useEffect, useState } from 'react';
+import { Props, TopSitesList } from './types';
+import RectSiteBox from './RectSiteBox';
+import './TopSites.sass';
+
+export const TopSites: FC<Props> = () => {
+    const namespace: typeof chrome | typeof browser = chrome === undefined ? browser : chrome;
+    const [topSites, setTopSites] = useState<TopSitesList>([]);
```

Comment:
```suggestion
export const TopSites: FC<Props> = ({ cache, setCache }) => {
```
If you use `cache` instead of `useState`, the cached top sites will be available on first render, without needing to wait for the promise to complete.